docs/architecture/accounts.md

```diff
@@ -1,211 +1,139 @@
+The `nonce` enforces ordering and prevents replay attacks or double-spending. It must strictly increase with every account state update. The increment must be less than `2^32` but always greater than the previous nonce, ensuring a well-defined sequence of state changes.
 
-      # store and drop the ASSET
-      padw mem_loadw.3 movup.4 exec.tx::add_asset_to_note
-      # => [note_ptr, ASSET, ZERO, ...]
-  end
+### Vault
 
-  #! Burns fungible assets.
-  #!
-  #! ...
-  export.burn
-      # burning the asset
-      exec.faucet::burn
-      # => [ASSET]
+> A collection of [assets](assets.md) stored by the `Account`.
 
-      # increments the nonce (anyone should be able to call that function)
-      push.1 exec.account::incr_nonce
+Assets are stored in a sparse Merkle tree, allowing for efficient, cryptographically secure proofs of ownership and state:
 
-      # clear the stack
-      padw swapw dropw
-      # => [...]
-  end
-  ```
-</details>
+- **Fungible assets:** Indexed by the issuing faucet ID. Each fungible asset type occupies exactly one node in the tree.  
+- **Non-fungible assets (NFTs):** Indexed by a unique asset identifier. Each NFT is represented by a distinct node in the tree.
 
-The contract exposes two functions `distribute` and `burn`.
+This arrangement facilitates efficient queries, updates, and proofs on assets.
 
-The first function, `distribute`, can only be called by the faucet owner; otherwise, it fails. As inputs, the function expects everything that is needed to create a note containing the freshly minted asset, i.e., amount, metadata, and recipient.
+### Code
 
-The second function, `burn`, burns the tokens that are contained in a note and can be called by anyone.
+> A collection of functions defining the `Account`’s programmable interface.
 
-> **Info: Difference between `burn` and `distribute`**
-> The `burn` procedure exposes `exec.account::incr_nonce`, so by calling `burn`, the nonce of the executing account gets increased by `1`, and the transaction will pass the epilogue check. The `distribute` procedure does not expose that. That means the executing user needs to call `basic::auth_tx_rpo_falcon512`, which requires the private key.
+Every Miden account is essentially a smart contract. The `Code` component defines the account’s functions, which can be invoked through both [Note scripts](notes.md#the-note-script) and transaction scripts. Key characteristics include:
 
-## Account creation
+- **Function commitment:** Each function corresponds to a root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html), represented as a 32-byte hash. This ensures both integrity and immutability of the underlying function logic.  
+- **Mutable access:** Only the account’s own functions can modify its storage and vault. All state changes—such as updating storage slots, incrementing the nonce, or transferring assets—must occur through these functions.  
```

Comment:
Can you bring the "Mutable access" bullet on top?

Reply:
can you change the "function commitment" bullet? I think it should be:

- **Function commitment:** Each function can be called by its [MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) root. The root represents the underlying code tree as a 32-byte hash. This ensures integrity, i.e., the caller calls what he expects.